src/shim.d.ts

```diff
@@ -0,0 +1,3 @@
+declare namespace Intl {
+  function getCanonicalLocales(locales: string | string[]): string[]
```

Comment:
Curretnly, `Intl.getCanonicalLocales` is type ineffective in typescript
So, I used shim to work around it. Maybe send a PR to TypeScript.

Reply:
related issues:
- https://github.com/microsoft/TypeScript/issues/29129
- https://github.com/microsoft/TypeScript/pull/45647
- https://github.com/microsoft/TypeScript/pull/39662